test/distributed/org/apache/cassandra/distributed/test/ReadRepairTest.java

```diff
@@ -64,14 +76,87 @@ public void readRepairTest() throws Throwable
         }
     }
 
+    /**
+     * Migrated from Python dtest consistency_test.py:TestConsistency.test_readrepair
```

Comment:
Not sure we need this migrated test, since `readRepairTest` does the same with a single row, and the 10000 rows in the original Python dtest don't seem to add any value.

Reply:
@adelapena I don't like it much either. It seems pretty arbitrary, uses QUORUM when there are only 2 nodes, etc.

What if we replaced it with a randomized/QuickTheories test w/ 3 nodes that did a hundred or so iterations of writing to 2 of the 3 nodes, then verifying the the 3rd node sees the data after the blocking RR for a QUORUM read? That could actually replace both this test and `readRepairTest()` and we could call it `testBlockingReadRepair()`. (Probably best to explicitly set `read_repair='BLOCKING'`.)

Reply:
It might also be nice to have a simple, non-randomized test for `read_repair='NONE'` just as a guardrail for someone accidentally breaking that bit of configuration. (If blocking RR were actually broken, another test would point that out.)

Reply:
...and a third thought that sort of brings this together is that w/ a couple properly designed randomized/QT tests here in `ReadRepairTest`, I think we could actually eliminate `ReadRepairGuaranteesTest` altogether. That would also mean we could ditch ByteBuddy, which was necessary for a literal port, but doesn't actually seem necessary to verify what we need to verify.

The usage of "atomic" is also a little confusing to me in `ReadRepairGuaranteesTest`, given it just seems to mean that "QUORUM reads aren't guaranteed to see values written to a single node". (i.e. With `NONE`, what exactly is "atomic" given that we can see the result of a write being only partially applied to the replica set?)

Reply:
I have just replaced both tests by a couple of per-strategy tests using QuickTheories and three nodes.

Reply:
I'm still not sure a.) `ReadRepairGuaranteesTest` covers anything `ReadRepairTest` does not (seems like the former just uses more complex machinery to make the same assertions on the same 2 basic cases) or b.) that randomizing the cell values themselves is useful in terms of finding potential problems in read repair.

Looking at this again, I think the only reason I suggested a QT test was I thought there was actually non-trivial parameterization on `ReadRepairGuaranteesTest`. Apologies.

Reply:
I have removed `ReadRepairGuaranteesTest` and the usage of QuickTheories in `ReadRepairTest#testBlockingReadRepair` and `ReadRepairTest#testNoneReadRepair` to keep them simple. WDYT?